docs/proposals/tenant_retention.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "Retention of Tenant Data from Blocks Storage"
+linkTitle: "Retention of Tenant Data from Blocks Storage"
+weight: 1
+slug: tenant-retention
+---
+
+- Author: [Allenzhli](https://github.com/Allenzhli)
+- Date: January 2021
+- Status: Proposed
+
+## Retention of tenant data
+
+## Problem
+
+Metric data is growing over time per-tenant, at the same time, the value of data decreases. We want to have a retention policy like prometheus does. In Cortex, data retention is typically achieved via a bucket policy. However, this has two main issues:
+
+1. Not every backend storage support bucket policies
+2. Bucket policies don't easily allow a per-tenant custom retention
+
+## Background
+
+### tenants
+When using blocks storage, Cortex stores tenant’s data in object store for long-term storage of blocks, tenant id as part of the object store path. We discover all tenants via scan the root dir of bucket.
+
+### runtime config
+Using the "overrides" mechanism (part of runtime config) already allows for per-tenant settings. See [runtime-configuration-file](https://cortexmetrics.io/docs/configuration/arguments/#runtime-configuration-file) for more details. Using it for tenant retention would fit nicely. Admin could set per-tenant retention here, and also have a single global value for tenants that don't have custom value set.
+
+## Proposal
+
+### retention period field
+
+We propose to introduce just one new field `RetentionPeriodPerTenant` in the Limits struct(defined at pkg/util/validation/limits.go).
```

Comment:
All fields in the `Limits` struct are per-tenant so no need to make it explicit.

```suggestion
We propose to introduce just one new field `RetentionPeriod` in the Limits struct(defined at pkg/util/validation/limits.go).
```